src/storage/data_table.cpp

```diff
@@ -172,14 +171,24 @@ ItemPointer DataTable::GetEmptyTupleSlot(const storage::Tuple *tuple) {
   // check if there are recycled tuple slots
   auto &gc_manager = gc::GCManagerFactory::GetInstance();
   auto free_item_pointer = gc_manager.ReturnFreeSlot(this->table_oid);
+  bool valid_slot = true;
+  ;
   if (free_item_pointer.IsNull() == false) {
     // when inserting a tuple
-    if (tuple != nullptr) {
+    if ((tuple != nullptr)) {
       auto tile_group =
           catalog::Manager::GetInstance().GetTileGroup(free_item_pointer.block);
-      tile_group->CopyTuple(tuple, free_item_pointer.offset);
+
+      if (tile_group->GetCompressionStatus() == true) {
+        valid_slot = false;
```

Comment:
No, returning at this point would return an invalid pointer to the calling executor function. Once valid_slot is set to false, the data_table would create a new empty slot (by allocating a new tile_group)  and return it to the executor.